include/jemalloc/internal/thread_event.h

```diff
@@ -0,0 +1,61 @@
+#ifndef JEMALLOC_INTERNAL_THREAD_EVENT_H
+#define JEMALLOC_INTERNAL_THREAD_EVENT_H
+
+#include "jemalloc/internal/tsd.h"
+
+/*
+ * Maximum threshold on thread_allocated_threshold_fast, so that there is no
+ * need to check overflow in malloc fast path. (The allocation size in malloc
+ * fast path never exceeds SC_LOOKUP_MAXCLASS.)
+ */
+#define THREAD_ALLOCATED_THRESHOLD_FAST_MAX				\
+    (UINT64_MAX - SC_LOOKUP_MAXCLASS + 1)
+
+/*
+ * Maximum threshold on wait time, so that whenever an event is triggered, the
+ * accumulated allocation bytes do not overflow uint64_t.
+ */
+#define THREAD_EVENT_WAIT_MAX (UINT64_MAX - SC_LARGE_MAXCLASS + 1)
```

Comment:
Why this can't be UINT64_MAX? which overflow will it hurt?

Reply:
the `bulk_transfer` - whenever an event is triggered, we are transferring `thread_allocated + usize - last_event` to the wait times, rather than just the last wait time.

Reply:
Discussed offline, we should be able to get rid of the limit, with one additional check in event_trigger.